### PR TITLE
fix: revert changes which broke route registration

### PR DESCRIPTION
**Pull request by denis-anisimov**
Need to active the tracker from an servlet context initializer because @Activate doesn't work on CTOR with parameter

### CODE REVIEW

**Review by pleku — CHANGES_REQUESTED** [review]
Can you please elaborate, even in the commit description line on the _why it was broken_ or how adding (back?) a servlet context initializer to activate the bundle tracker is necessary. Something along the lines "Need to active the tracker because foobar"

File: flow-server/src/main/java/com/vaadin/flow/server/osgi/VaadinBundleTracker.java [hunk 1]
```diff
@@ -65,7 +63,6 @@
  * @author Vaadin Ltd
  * @since 1.2
  */
-@Component(immediate = true)
```
[the remark this answers is not in the document]
Comment (denis-anisimov) [reply flow-server/src/main/java/com/vaadin/flow/server/osgi/VaadinBundleTracker.java:68]: Apparently it never worked.
I don't know why it worked for me when I tried it.
May be it's related to the versions and then which version : bnd or OSGi container/depdencies.
I don't know.

Currently it doesn't work for sure. And in fact this is absolutely not important : one way against another is just a convenience. Stefan will hopefully find a reason and fix this minor thing. There are much more important things now.

**Comment by stbischof** [comment]
What does not work? could you please provide a test here that shows the problem?

**Comment by denis-anisimov** [comment]
> What does not work? could you please provide a test here that shows the problem?

Please wait for the validation in OSGi repository.
Currently it's broken and this will fix it.
Until we have a validation enabled any discussion is a waste of time.

If you want this _now_ then please check https://github.com/vaadin/osgi/tree/master/test-containers/felix-jetty by yourself.
The tests there are failing because routes are not registered.

Nothing should be done until validation is enabled: I don't want to be a proxy for this validation.

**Comment by stbischof** [comment]
I don't know about the `vaadin/osgi/` Project
and do not have any access to
https://github.com/vaadin/osgi/tree/master/test-containers/felix-jetty

**Comment by denis-anisimov** [comment]
> I don't know about the `vaadin/osgi/` Project
> and do not have any access to
> https://github.com/vaadin/osgi/tree/master/test-containers/felix-jetty

Ops, sorry. It turns out it's private.
I can make it public but I should ask first.
Let's postpone this till tomorrow.

**Comment by denis-anisimov** [comment]
@stbischof , the repository is public now.

**Review by pleku — CHANGES_REQUESTED** [review]
Like SQ says, it is missing javadocs

**Comment by denis-anisimov** [comment]
> Like SQ says, it is missing javadocs

SQ doesn't know about `@Activate`/`@Deactivate` annotations.
I will add totally useless  comments which just repeats the same as annotations mean because of their semantic.
But this has no any sense.

**Review by pleku — DISMISSED** [review]

**Comment by pleku** [comment]
>SQ doesn't know about @Activate/@Deactivate annotations.
I will add totally useless comments which just repeats the same as annotations mean because of their semantic.
But this has no any sense.

This is the fact for most of the javadocs but we still keep on writing them, so I don't see a point in complaining in something that we've agreed on doing for years and for good reasons.

**Comment by denis-anisimov** [comment]
> > SQ doesn't know about @Activate/@deactivate annotations.
> > I will add totally useless comments which just repeats the same as annotations mean because of their semantic.
> > But this has no any sense.
> 
> This is the fact for most of the javadocs but we still keep on writing them, so I don't see a point in complaining in something that we've agreed on doing for years and for good reasons.

The point here is : these methods are equivalent of @Overrides for some interface method which already exists in some interface and documented. Annotation is just another way for this. And we don't repeat javadocs in these cases.

And in fact: I'm strongly against of having dumb comments. E.g. I don't see any sense to write a comment for default CTOR:
- extra activity 
- you don't write anything useful there
- this adds a number of extra lines

On the other hand : yes it's something that we silently agreed and may be it's time to reconsider this  generally.

**Comment by pleku** [comment]
>On the other hand : yes it's something that we silently agreed and may be it's time to reconsider this generally.

If know how to make it so that we can easily enforce when the javadocs are needed and when not then please share. Otherwise there are soon (this is a fact) methods that would need javadocs but don't have them because not required out-of-the-box.

**Comment by denis-anisimov** [comment]
> > On the other hand : yes it's something that we silently agreed and may be it's time to reconsider this generally.
> 
> If know how to make it so that we can easily enforce when the javadocs are needed and when not then please share. Otherwise there are soon (this is a fact) methods that would need javadocs but don't have them because not required out-of-the-box.

I'm pretty sure that SQ can't be configured to understand when javadocs are not required.
Semantically we already have this rules which SQ doesn't recognize in all cases:
- private classes don't need javadocs : neither class level nor methods level regardless of their visibility
- overriden methods don't need javadocs.

And I would add : default CTOR (without parameters) don't need javadocs because it's always dumb: "Create a new instance".

Now the rules could be extended to semantic instead only static Java rules:
- `ServletContainerInitializerExtender` is effectively private . It's not private just because it's registered via DS and it's a framework limitation that it should be `public` . I would may create an activator class (again: `public` and there is no any sense to document the activator) and register a private `ServletContainerInitializerExtender`  class programatically. So this class if effectively private implementation details which is exposed only because of DS registration
- methods which are implementation of some contract (e.g. as here via annotations) should be treated the same as `@Overrides`  and don't need documentation

And default CTOR  doesn't need documentation.
Even this should be already cover most cases.

**Review by pleku — APPROVED** [review]

**Comment by pleku** [comment]
> And I would add : default CTOR (without parameters) don't need javadocs because it's always dumb: "Create a new instance".

Probably most of the time it is so, but not when there is specific semantic in it compared to the other constructors. Probably quite rare.

If we cannot figure out a way to improve SQ then we should agree as a team to add something else instead for the cases
- no javadocs needed for default constructor
- if something is effectively private
One option could be to supress the complaints
```java 
@SuppressWarnings("all") // nothing is warned about

// only specific rules
@SuppressWarnings("squid:S2078") 
@SuppressWarnings({"squid:S2078", "squid:S2076"})
```
but I think this would take as much time to write (even with the rules as constants somewhere) than just writing the silly javadoc.

**Pull request merged**